test/contracts/provider/setup_test.go

```diff
@@ -0,0 +1,263 @@
+package provider_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pmacik/loginusers-go/config"
+	"github.com/pmacik/loginusers-go/loginusers"
+)
+
+type providerStateInfo struct {
+	// Consumer name
+	Consumer string `json:"consumer"`
+	// State
+	State string `json:"state"`
+	// States
+	States []string `json:"states"`
+}
+
+type ProviderInitialState struct {
+	User   User
+	Tokens loginusers.Tokens
+}
+
+type createUserAttributes struct {
+	Bio       string `json:"bio"`
+	Cluster   string `json:"cluster"`
+	Username  string `json:"username"`
+	Email     string `json:"email"`
+	RhdUserID string `json:"rhd_user_id"`
+}
+
+type createUserData struct {
```

Comment:
I think we can use it from client, no need to define it multiple times, this way it's easy to maintain, https://github.com/fabric8-services/fabric8-auth-client/blob/master/auth/user_types.go#L351

Reply:
Switched to auth-client.